src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/ObjectPools/Extensions.cs

```diff
@@ -285,7 +285,7 @@ public static void ClearAndFree<T>(this ObjectPool<List<T>> pool, List<T> list)
             pool.Free(list);
         }
 
-        public static void ClearAndFree<T>(this ObjectPool<SegmentedList<T>> pool, SegmentedList<T> list)
+        public static void ClearAndFree<T>(this ObjectPool<SegmentedList<T>> pool, SegmentedList<T> list, bool trim = true)
```

Comment:
>public static void ClearAndFree<T>(this ObjectPool<SegmentedList<T>> pool, SegmentedList<T> list, bool trim = true) [](http://example.com/codeflow?start=8&length=115)

I assume it's preferable to minimize the changes in this PR rather than modify the other ClearAndFree methods similarly?

Reply:
Yup!